code/parachain/frame/composable-traits/src/currency.rs

```diff
@@ -142,13 +154,93 @@ impl<
 {
 }
 
-// hack to imitate type alias until it is in stable
-// named with like implying it is`like` is is necessary to be `AssetId`, but may be not enough (if
-// something is `AssetIdLike` than it is not always asset)
+pub trait AssetIdLike = FullCodec + MaxEncodedLen + Copy + Eq + PartialEq + Debug + TypeInfo;
+
+#[derive(
+	RuntimeDebug,
+	Copy,
+	Clone,
+	PartialEq,
+	Eq,
+	PartialOrd,
+	Ord,
+	Encode,
+	Decode,
+	MaxEncodedLen,
+	TypeInfo,
+)]
+pub struct Rational64 {
+	pub numer: u64,
+	pub denom: u64,
+}
 
-// FIXME(hussein-aitlahcen): this trait already exists in frame_support, named `AssetId`
-pub trait AssetIdLike:
-	FullCodec + MaxEncodedLen + Copy + Eq + PartialEq + Debug + TypeInfo
-{
+#[macro_export]
+macro_rules! rational {
+	($numer:literal / $denom: literal) => {
+		Rational64 { numer: $numer, denom: $denom }
+	};
+}
+
+macro_rules! impl_const_get {
+	($name:ident, $t:ty) => {
+		#[doc = "Const getter for a basic type."]
+		#[derive(RuntimeDebug)]
+		pub struct $name<const T: $t>;
+		impl<const T: $t> Get<$t> for $name<T> {
+			fn get() -> $t {
+				T
+			}
+		}
+		impl<const T: $t> Get<Option<$t>> for $name<T> {
+			fn get() -> Option<$t> {
+				Some(T)
+			}
+		}
+		impl<const T: $t> TypedGet for $name<T> {
+			type Type = $t;
+			fn get() -> $t {
+				T
+			}
+		}
+	};
+}
+
+impl_const_get!(ConstRational64, Rational64);
+
+impl Rational64 {
+	pub fn new(numer: u64, denom: u64) -> Self {
+		Rational64 { numer, denom }
+	}
```

Comment:
For this an all other creations of `Rational*` types, we should be checking that `denom` is never `0`.